### PR TITLE
Added logic for case when detector is armed prior to scan

### PR DESCRIPTION
**Pull request by RobertSchaffer1**

### CODE REVIEW

**Comment by RobertSchaffer1** [comment]
That looks a bit healthier...

File: daq_macros.py [hunk 1]
```diff
@@ -1840,6 +1840,14 @@ def snakeRasterBluesky(rasterReqID, grain=""):
...
 
     raster_flyer.configure_detector(file_prefix=rasterFilePrefix, data_directory_name=data_directory_name)
+    if raster_flyer.detector.cam.armed.get() == 1:
+        daq_lib.gui_message('Detector is in armed state from previous collection! Stopping detector, but the user '
+                            'should check the most recent collection to determine if it was successful. Cancelling'
+                            'this collection, retry when ready.')
+        raster_flyer.detector.cam.acquire.put(0)
+        logger.warning("Detector was in the armed state prior to this attempted collection.")
+        return 0
+        #raise RuntimeError("Detector is armed, cannot continue in current state.")
```
Comment (JunAishima) [inline daq_macros.py:1850]: can you get rid of this last commented-out line?

**Review by JunAishima — CHANGES_REQUESTED** [review]
please just remove that last line and otherwise looks good!

strictly speaking, stopping the detector won't be necessary because we are using the finalize_wrapper that will always stop it at the end of the function call, but it's good to see anyways.

File: daq_macros.py (hunk 1 above)
Comment (RobertSchaffer1) [reply daq_macros.py]: Done.

**Review by JunAishima — APPROVED** [review]

**Pull request merged**